azurerm/internal/services/loganalytics/log_analytics_datasource_linux_syslog_resource.go

```diff
@@ -0,0 +1,226 @@
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmLogAnalyticsDataSourceLinuxSyslogRead(d, meta)
+}
+
+func resourceArmLogAnalyticsDataSourceLinuxSyslogRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).LogAnalytics.DataSourcesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.LogAnalyticsDataSourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Workspace, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Log Analytics DataSource Linux syslog %q was not found in Resource Group %q in Workspace %q - removing from state!", id.Name, id.ResourceGroup, id.Workspace)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("failed to retrieve Log Analytics DataSource Linux syslog %q (Resource Group %q / Workspace: %q): %+v", id.Name, id.ResourceGroup, id.Workspace, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("workspace_name", id.Workspace)
+	if props := resp.Properties; props != nil {
+		propStr, err := structure.FlattenJsonToString(props.(map[string]interface{}))
+		if err != nil {
+			return fmt.Errorf("failed to flatten properties map to json for Log Analytics DataSource Linux syslog %q (Resource Group %q / Workspace: %q): %+v", id.Name, id.ResourceGroup, id.Workspace, err)
+		}
+
+		prop := &dataSourceLinuxSysLogProperty{}
+		if err := json.Unmarshal([]byte(propStr), &prop); err != nil {
+			return fmt.Errorf("failed to decode properties json for Log Analytics DataSource Linux syslog %q (Resource Group %q / Workspace: %q): %+v", id.Name, id.ResourceGroup, id.Workspace, err)
+		}
+
+		d.Set("syslog_name", prop.SysLogName)
+		d.Set("syslog_severities", flattenLogAnalyticsDataSourceLinuxSyslogSeverity(prop.SysLogSeverities))
+	}
+
+	return nil
+}
+
+func resourceArmLogAnalyticsDataSourceLinuxSyslogDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).LogAnalytics.DataSourcesClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.LogAnalyticsDataSourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	if _, err := client.Delete(ctx, id.ResourceGroup, id.Workspace, id.Name); err != nil {
+		return fmt.Errorf("failed to delete Log Analytics DataSource Linux syslog %q (Resource Group %q / Workspace: %q): %+v", id.Name, id.ResourceGroup, id.Workspace, err)
+	}
+
+	return nil
+}
+
+func expandLogAnalyticsDataSourceLinuxSyslogSeverity(input []interface{}) []dataSourceLinuxSysLogSeverity {
+	output := []dataSourceLinuxSysLogSeverity{}
+	for _, severity := range input {
+		output = append(output, dataSourceLinuxSysLogSeverity{severity.(string)})
+	}
+	return output
+}
+
+func flattenLogAnalyticsDataSourceLinuxSyslogSeverity(severities []dataSourceLinuxSysLogSeverity) []interface{} {
```

Comment:
Same here, can we add a nil or empty value check here?

Reply:
Added